finagle-core/src/main/scala/com/twitter/finagle/util/LoadService.scala

```diff
@@ -68,17 +68,22 @@ private object ClassPath {
     ents
   }
 
-  private[finagle] def browseUri(uri: URI, loader: ClassLoader, buf: mutable.Buffer[Info]) {
+  private[finagle] def browseUri(uri: URI, loader: ClassLoader, buf: mutable.Buffer[Info], history: mutable.Set[String] = mutable.Set[String]()) {
     if (uri.getScheme != "file")
       return
     val f = new File(uri)
     if (!(f.exists() && f.canRead))
       return
+    val path = f.getCanonicalPath
```

Comment:
instead of turning these into strings, why not just keep them `File`s?  Java supports rich equality / hashCode for them, so they also allow things like checking if two pathnames are the same on case-insensitive filesystems.